core/src/main/java/discord4j/core/object/ExtendedInvite.java

```diff
@@ -66,13 +66,20 @@ public int getMaxUses() {
     /**
      * Gets the instant this invite expires, if possible.
      *
-     * @return The instant this invite expires, if possible.
+     * @return The instant this invite expires, if empty, invite is never expiring.
      */
     public Optional<Instant> getExpiration() {
-        final boolean temporary = getData().temporary().toOptional().orElse(false);
         final int maxAge = getData().maxAge().toOptional().orElseThrow(IllegalStateException::new);
+        return maxAge > 0 ? Optional.of(getCreation().plus(maxAge, ChronoUnit.SECONDS)): Optional.empty();
```

Comment:
This file needs a reformat code action, for this line and also indentation at isTemporary()